Source/WebKit/UIProcess/mac/PageClientImplMac.mm

```diff
@@ -293,6 +297,21 @@ - (BOOL)_hostsLayersInWindowServer;
 #if ENABLE(WRITING_TOOLS)
     m_impl->hideTextAnimationView();
 #endif
+
+#if ENABLE(SCREEN_TIME)
+    if (!PAL::isScreenTimeFrameworkAvailable())
+        return;
```

Comment:
This early return seems a bit dangerous for people adding unrelated code later in this method. Maybe just nest it and invert the if? Or factor it out into another method instead of nesting? I'm not sure.

Reply:
I think factoring out this logic into another method works better. Early returning in that separate method will not effect code after in `didCommitLoadForMainFrame`! Will change!